packages/ai/core/embed/embed-many.ts

```diff
@@ -104,7 +104,7 @@ Only applicable for HTTP-based providers.
 
       // the model has not specified limits on
       // how many embeddings can be generated in a single call
-      if (maxEmbeddingsPerCall == null) {
+      if (maxEmbeddingsPerCall == null || maxEmbeddingsPerCall === Infinity) {
```

Comment:
wouldn't we need to check if it's a promise and optionally await?

Reply:
ooops